lisa/tools/ethtool.py

```diff
@@ -0,0 +1,181 @@
+    )
+
+    def __init__(self, device: str, device_channel_raw: str) -> None:
+        self.parse_channel_info(device, device_channel_raw)
+
+    def parse_channel_info(self, device: str, raw_str: str) -> None:
+        current_settings = self.__current_settings_pattern.match(raw_str)
+        max_settings = self.__max_settings_pattern.match(raw_str)
+        if (not current_settings) or (not max_settings):
+            raise LisaException(
+                "cannot get device channel current and/or max settings information"
+            )
+
+        current_param = self.__channel_count_param_pattern.match(
+            current_settings.group("settings")
+        )
+        max_param = self.__channel_count_param_pattern.match(
+            max_settings.group("settings")
+        )
+        if (not current_param) or (not max_param):
+            raise LisaException("Cannot get device channel current and/or max count")
+
+        self.device_name = device
+        self.current_channels = int(current_param.group("value"))
+        self.max_channels = int(max_param.group("value"))
+
+
+class DeviceFeatures:
+    __feature_info_pattern = re.compile(
+        r"Features for (?P<interface>[\w]*):[\s]*(?P<value>.*?)$", re.DOTALL
+    )
+
+    __feature_settings_pattern = re.compile(
+        r"^[\s]*(?P<name>.*):(?P<value>.*?)?$", re.MULTILINE
+    )
+
+    def __init__(self, device: str, device_feature_raw: str) -> None:
+        self.parse_feature_info(device, device_feature_raw)
+
+    def parse_feature_info(self, device: str, raw_str: str) -> None:
+        matched_features_info = self.__feature_info_pattern.match(raw_str)
+        if not matched_features_info:
+            raise LisaException("cannot get device's features settings info")
+
+        self.device_name = device
+        self.enabled_features = []
+        for row in matched_features_info.group("value").splitlines():
+            feature_info = self.__feature_settings_pattern.match(row)
+            if not feature_info:
+                raise LisaException()
+            if "on" in feature_info.group("value"):
+                self.enabled_features.append(feature_info.group("name"))
+
+
+class Ethtool(Tool):
+    @property
+    def command(self) -> str:
+        return self._command
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        self._command = "ethtool"
+        self._device_list: List[str] = []
+        self._device_channel: List[DeviceChannel] = []
+        self._device_features: List[DeviceFeatures] = []
+
+    def _install(self) -> bool:
+        posix_os: Posix = cast(Posix, self.node.os)
+        posix_os.install_packages("ethtool")
+        return self._check_exists()
+
+    def _get_device_list(self) -> None:
+        if self._device_list:
+            return
+
+        netdirs = list(Path("/sys/devices/").rglob("*VMBUS*/**/net*"))
```

Comment:
Is it run on local only? It should be executed from node to support remote node.

Reply:
It should use PurePath

Reply:
Why PurePath is necessary here?

Reply:
PurePath is fit to both Linux and Windows. Although this logic most likely doesn't fit to Windows, but it's good practice to use PurePath on remote paths.
But it needs to be called form node to support remote node.